app/models/blood_pressure.rb

```diff
@@ -5,9 +5,12 @@ class BloodPressure < ApplicationRecord
   ANONYMIZED_DATA_FIELDS = %w[id patient_id created_at bp_date registration_facility_name user_id
                               bp_systolic bp_diastolic]
 
-  belongs_to :facility, optional: true
   belongs_to :patient, optional: true
   belongs_to :user, optional: true
+  belongs_to :facility, optional: true
+
+  has_one :encounter_event, as: :encounterable
```

Comment:
Perhaps we should call these `Observation` to match FHIR?

Reply:
`Observations` is a much better name than `Event`. Thanks!